### ISSUE TITLE
[gcp] v0.3.0-rc.1 cert manager can't get the SSL certificate

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#1666]
When I started Kubeflow on GCP with v0.3.0-rc.1, certmanager is having problems.

```
E0929 19:50:10.497029       1 reflector.go:205] github.com/jetstack/cert-manager/pkg/client/informers/externalversions/factory.go:71: Failed to list *v1alpha1.Issuer: v1alpha1.IssuerList: Items: []v1alpha1.Issuer: v1alpha1.Issuer: Spec: v1alpha1.IssuerSpec: IssuerConfig: ACME: v1alpha1.ACMEIssuer: HTTP01: skipObjectDecoder: expect object or null, error found in #10 byte of ...|,"http01":"","privat|..., bigger context ...|pec":{"acme":{"email":"jlewi@google.com","http01":"","privateKeySecretRef":{"name":"letsencrypt-prod|...

```

Related to #1541

**Comment by jlewi** [comment]
I suspect this was fixed in #1580 and the changes need to be cherry picked onto the release branch.

**Comment by jlewi** [comment]
Should be fixed in latest RC.
